server/datastore/mysql/scripts.go

```diff
@@ -737,40 +843,76 @@ func (ds *Datastore) UnlockHostManually(ctx context.Context, hostID uint, ts tim
 	// entering a PIN on the device). The /unlock endpoint can be called multiple
 	// times, so we record the timestamp of the first time it was requested and
 	// from then on, the host is marked as "pending unlock" until the device is
-	// actually unlocked with the PIN.
-	// TODO(mna): to be determined how we then get notified that it has been
-	// unlocked, so that it can transition to unlocked (not pending).
+	// actually unlocked with the PIN. The actual unlocking happens when the
+	// device sends an Idle MDM request.
 	unlockRef := ts.Format(time.DateTime)
 	_, err := ds.writer(ctx).ExecContext(ctx, stmt, hostID, unlockRef)
 	return ctxerr.Wrap(ctx, err, "record manual unlock host request")
 }
 
-func (ds *Datastore) updateHostLockWipeStatusFromResult(ctx context.Context, tx sqlx.ExtContext, hostID uint, refCol string, succeeded bool) error {
-	stmt := `UPDATE host_mdm_actions SET %s WHERE host_id = ?`
+func buildHostLockWipeStatusUpdateStmt(refCol string, succeeded bool, joinPart string) string {
+	var alias string
+
+	stmt := `UPDATE host_mdm_actions `
+	if joinPart != "" {
+		stmt += `hma ` + joinPart
+		alias = "hma."
+	}
+	stmt += ` SET `
 
 	if succeeded {
 		switch refCol {
 		case "lock_ref":
 			// Note that this must not clear the unlock_pin, because recording the
 			// lock request does generate the PIN and store it there to be used by an
 			// eventual unlock.
-			stmt = fmt.Sprintf(stmt, "unlock_ref = NULL")
+			stmt += fmt.Sprintf("%sunlock_ref = NULL, %[1]swipe_ref = NULL", alias)
 		case "unlock_ref":
 			// a successful unlock clears itself as well as the lock ref, because
 			// unlock is the default state so we don't need to keep its unlock_ref
 			// around once it's confirmed.
-			stmt = fmt.Sprintf(stmt, "lock_ref = NULL, unlock_ref = NULL, unlock_pin = NULL")
+			stmt += fmt.Sprintf("%slock_ref = NULL, %[1]sunlock_ref = NULL, %[1]sunlock_pin = NULL, %[1]swipe_ref = NULL", alias)
 		case "wipe_ref":
-			// TODO(mna): implement when implementing the wipe story
-		default:
-			return ctxerr.Errorf(ctx, "unknown reference column %q", refCol)
+			// TODO(mna): can a wiped device still be locked? If so we'd need to keep
+			// the pin/lock_ref, but this would mean that both IsLocked and IsWiped
+			// would be true, we'd need to ensure we check wipe first (I think we
+			// always assumed those 3 states were exclusive).
```

Comment:
See https://github.com/fleetdm/fleet/issues/9951#issuecomment-1955108805

Reply:
Looks like we're waiting on some feedback from Noah: https://github.com/fleetdm/fleet/issues/9951#issuecomment-1960015916

Reply:
I think we can go ahead like this for now, as this is likely the intention (wipe only if unlocked):

> I think the current plan is to allow it when the host is unlocked (what we have in the UI wireframes)

We can address in a subsequent PR if this needs to change.